api/caasoperator/client_test.go

```diff
@@ -166,3 +167,40 @@ func (s *operatorSuite) TestWatchApplicationConfig(c *gc.C) {
 	c.Assert(watcher, gc.IsNil)
 	c.Assert(err, gc.ErrorMatches, "FAIL")
 }
+
+func (s *operatorSuite) TestSetContainerSpec(c *gc.C) {
+	s.testSetContainerSpec(c, names.NewApplicationTag("gitlab"))
+	s.testSetContainerSpec(c, names.NewUnitTag("gitlab/0"))
+}
+
+func (s *operatorSuite) testSetContainerSpec(c *gc.C, tag names.Tag) {
+	apiCaller := basetesting.APICallerFunc(func(objType string, version int, id, request string, arg, result interface{}) error {
+		c.Check(objType, gc.Equals, "CAASOperator")
+		c.Check(version, gc.Equals, 0)
+		c.Check(id, gc.Equals, "")
+		c.Check(request, gc.Equals, "SetContainerSpec")
+		c.Check(arg, jc.DeepEquals, params.SetContainerSpecParams{
+			Entities: []params.EntityString{{
+				Tag:   tag.String(),
+				Value: "spec",
+			}},
+		})
+		c.Assert(result, gc.FitsTypeOf, &params.ErrorResults{})
+		*(result.(*params.ErrorResults)) = params.ErrorResults{
+			Results: []params.ErrorResult{{Error: &params.Error{Message: "bletch"}}},
+		}
+		return nil
+	})
+
+	client := caasoperator.NewClient(apiCaller)
+	err := client.SetContainerSpec(tag.Id(), "spec")
+	c.Assert(err, gc.ErrorMatches, "bletch")
+}
+
+func (s *operatorSuite) TestSetContainerSpecInvalidEntityame(c *gc.C) {
+	client := caasoperator.NewClient(basetesting.APICallerFunc(func(_ string, _ int, _, _ string, _, _ interface{}) error {
+		return errors.New("should not be called")
```

Comment:
c.Fail()?
Guess it doesn't matter

Reply:
doesn't matter - it'll be clear why it fails, if it ever does